### PR TITLE
Implement core-data reading V2 API testcases

### LINKED ISSUE #279: Implement Core Data reading V2 API testcases
All reading V2 API:
GET /reading/all
GET /reading/resourceName/{resourceName}
GET /reading/device/name/{name}
GET /reading/count
GET /reading/count/device/name/{name}
GET /reading/start/{start}/end/{end}

### PR DESCRIPTION
**Pull request by jinlinGuan**
Fix #279

Signed-off-by: Ginny Guan <ginny@iotechsys.com>

### CODE REVIEW

File: TAF/testScenarios/functionalTest/V2-API/core-data/reading/GET-negative.robot [hunk 1]
```diff
@@ -1,38 +1,68 @@
...
+ErrReadingGET002 - Query all readings with with offset out of range
+    When Run Keyword And Expect Error  *  Query All Readings With offset=-1
+    Then Should Return Status Code "400"
+    And Should Return Content-Type "application/json"
     And Response Time Should Be Less Than "${default_response_time_threshold}"ms
 
-ErrReadingGET002 - Query all readings with specified device by device name fails
-    When Query All Readings With Specified Device By Invalid Device Name
-    Then Should Return Status Code "404"
+ErrReadingGET003 - Query all readings with with non-int value on limit
```
Comment (cherrycl) [inline TAF/testScenarios/functionalTest/V2-API/core-data/reading/GET-negative.robot:27]: Contains duplicate `with`.

File: TAF/testScenarios/functionalTest/V2-API/core-data/reading/GET-negative.robot [hunk 2]
```diff
@@ -1,38 +1,68 @@
...
+    When Run Keyword And Expect Error  *  Query All Readings With limit=50001
+    Then Should Return Status Code "400"
+    And Should Return Content-Type "application/json"
     And Response Time Should Be Less Than "${default_response_time_threshold}"ms
 
-ErrReadingGET003 - Query readings by start/end time fails (Invalid Start)
-    When Query Readings By Invalid Start Time
+ErrReadingGET004 - Query all readings with with limit out of range
```
Comment (cherrycl) [inline TAF/testScenarios/functionalTest/V2-API/core-data/reading/GET-negative.robot:33]: Contains duplicate with.

Comment (jinlinGuan) [reply TAF/testScenarios/functionalTest/V2-API/core-data/reading/GET-negative.robot]: Removed.

File: TAF/testScenarios/functionalTest/V2-API/core-data/reading/GET-negative.robot (hunk 1 above)
Comment (jinlinGuan) [reply TAF/testScenarios/functionalTest/V2-API/core-data/reading/GET-negative.robot]: Removed.

**Review by cherrycl — APPROVED** [review]
LGTM

**Pull request merged**